subsys/bluetooth/services/bas/bas_internal.h

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2024 Demant A/S
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef BT_BAS_INTERNAL_H_
+#define BT_BAS_INTERNAL_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <sys/types.h>
+#include <zephyr/sys/atomic.h>
+#include <zephyr/types.h>
+#include <zephyr/bluetooth/services/bas_bls.h>
+
+/**
+ * @brief   Battery Service data structure.
+ */
+typedef struct {
```

Comment:
Why a typedef rather than a a normal struct? We normally reserve typedefs mostly for places where the type should stay opaque and the members shouldn't be directly accessed.